client/components/domains/contact-details-form-fields/managed-contact-details-form-fields.jsx

```diff
@@ -181,6 +181,10 @@ export class ManagedContactDetailsFormFields extends Component {
 				this.handleFieldChange( 'postal-code', formattedPostalCode );
 			}
 		} );
+
+		// Strip leading and trailing whitespace
+		const sanitizedValue = form[ camelCase( name ) ]?.value.trim();
```

Comment:
I've added this because presumably there's a reason why old checkout had it, although not fully knowing how old checkout handled sanitization there's a possibility that this will behave differently from the user's point of view.

Specifically, if the user enters a field with diacritics (very common for names and place names) then this will remove them when the field blurs. What's awkward about this is that, regardless of any technical limitations imposed by a registrar, the text displayed back to the user will be _wrong_, and they won't be able to fix it.